### PR TITLE
data-dependencies: Call buildHiddenRefMap once per package.

### PR DESCRIPTION
**Pull request by associahedron**
Instead of calling it once per module. This cuts the compile time for a large data-dependency by 25s. To put it in perspective, the total build time goes from 3m52s to 3m27s, so it's a modest but not particularly noticeable difference.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs [hunk 1]
```diff
@@ -833,8 +840,8 @@ refModule = \case
 type RefGraph = HMS.HashMap Ref (Bool, [Ref])
 
 -- | Calculate the set of all references that should be hidden.
-buildHiddenRefMap :: Env -> HMS.HashMap Ref Bool
-buildHiddenRefMap env =
+buildHiddenRefMap :: Config -> LF.World -> HMS.HashMap Ref Bool
```
[the remark this answers is not in the document]
Comment (associahedron) [reply compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs]: Yes! I think that should be the next experiment :)

**Comment by associahedron** [comment]
I removed the special cases I had left in the last PR, since they didn't seem to make a difference either way for build performance.

File: compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (associahedron) [reply compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs]: Wait, it shouldn't make a difference unless you have multiple data-dependencies with lots of packages in common.

[the remark this answers is not in the document]
Comment (cocreature) [reply compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs]: Right but having multiple data-dependencies that are interlinked doesn’t sound that unreasonable.

[the remark this answers is not in the document]
Comment (associahedron) [reply compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs]: Yes, but perhaps that change can be made if/when it becomes relevant, and/or as part of a larger refactoring of packaging rules.

[the remark this answers is not in the document]
Comment (cocreature) [reply compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs]: :+1:

**Pull request merged**